Tests/ClientTests/Autofill/CreditCard/ViewComponents/SensitiveHostingControllerTests.swift

```diff
@@ -0,0 +1,50 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import SwiftUI
+import UIKit
+import XCTest
+@testable import Client
+
+class SensitiveHostingControllerTests: XCTestCaseRootViewController {
+    var mockNotificationCenter: SpyNotificationCenter!
```

Comment:
Heya, I had created `SpyNotificationCenter` before we had `NotificationProtocol`/`MockNotificationCenter` available. Orla created the later classes and they are better and easier to use for tests around notification protocol. I created [a task](https://mozilla-hub.atlassian.net/browse/FXIOS-5929) so we remove `SpyNotificationCenter`.

So if you inject a `NotificationProtocol` rather than the concrete `NotificationCenter` type in `SensitiveHostingController` you'll be able to use the `MockNotificationCenter`. This will enable to add proper asserts regarding notifications. I would asserts for `postCallCount`, `addObserverCallCount` when the view gets created and `removeObserverCallCount` when the view is deinit. 

Also note, since you now have access to `MockAppAuthenticator`, you can assert that blur gets removed/added, and that the variable `isAuthenticated` gets proper status depending on the authentication states. 
So we test the decisions made inside that view depending on the authentication status, and we check we have a blur/no blur when we expect it to.

Reply:
`NotificationProtocol` is now implemented 🎉 but for `isAuthenticated` and blur check on notification handle we had a couple of issues to add tests for.  Basically, calling the function wasn't calling the function, even if the object existed on tests. Either Xcode was messing with us, or we had a serious issue. No matter, the PR work is great. We can try to figure this out later!